cli/run.go

```diff
@@ -125,12 +131,31 @@ func (rc *RunCommand) runRun(args []string) error {
 		}
 		defer conn.Close()
 
+		var in io.Reader = os.Stdin
+		escapeKeys := defaultEscapeKeys
+		// Wrap the input to detect detach escape sequence.
+		// Use default escape keys if an invalid sequence is given.
+		if rc.detachKeys != "" {
+			customEscapeKeys, err := term.ToBytes(rc.detachKeys)
+			if err != nil {
+				logrus.Warnf("invalid detach escape keys, using default: %s", err)
+			} else {
+				escapeKeys = customEscapeKeys
+			}
+		}
+
+		in = ioutils.NewReadCloserWrapper(term.NewEscapeProxy(os.Stdin, escapeKeys), os.Stdin.Close)
+
 		go func() {
 			io.Copy(os.Stdout, br)
 			wait <- struct{}{}
 		}()
 		go func() {
-			io.Copy(conn, os.Stdin)
+			_, err := io.Copy(conn, in)
+			if _, ok := err.(term.EscapeError); ok {
```

Comment:
@xiaoxubeii could you mind to provide more detail about this kind of error? what condition will trigger this error?

for now, we cannot just send the signal to the channel to quit `run` command because we need to send the close stdin api to containerd. I provide more detail in the PR #2258. In my opinion, keep it. just print the error and return. we can fix it in the following PR. What do you think about this?

Reply:
> @xiaoxubeii could you mind to provide more detail about this kind of error? what condition will trigger this error?
> 
> for now, we cannot just send the signal to the channel to quit `run` command because we need to send the close stdin api to containerd. I provide more detail in the PR #2258. In my opinion, keep it. just print the error and return. we can fix it in the following PR. What do you think about this?

It will return `term.EscapeError` when catches a detach keys:
https://github.com/alibaba/pouch/blob/f285f1863903ba4bc0778b45d0d194786ce9e98a/vendor/github.com/docker/docker/pkg/term/proxy.go#L65-L67

OK, I would print the error instead of closing the io, we can fix it later.